vertexai/embeddings/generate_for_image.go

```diff
@@ -0,0 +1,89 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package embeddings shows examples of working with multimodal embeddings in Vertex AI
+package embeddings
+
+// [START generativeaionvertexai_multimodal_embedding_image]
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	aiplatform "cloud.google.com/go/aiplatform/apiv1beta1"
+	aiplatformpb "cloud.google.com/go/aiplatform/apiv1beta1/aiplatformpb"
+	"google.golang.org/api/option"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+// generateForImage shows how to use the multimodal model to generate embeddings for image input
+func generateForImage(w io.Writer, project, location string) ([]float32, error) {
+	// location = "us-central1"
+	ctx := context.Background()
+	apiEndpoint := fmt.Sprintf("%s-aiplatform.googleapis.com:443", location)
+	client, err := aiplatform.NewPredictionClient(ctx, option.WithEndpoint(apiEndpoint))
+	if err != nil {
+		return nil, fmt.Errorf("failed to construct API client: %v", err)
+	}
+	defer client.Close()
+
+	model := "multimodalembedding@001"
+	endpoint := fmt.Sprintf("projects/%s/locations/%s/publishers/google/models/%s", project, location, model)
+
+	// This is the input to the model's prediction call. The schema of any single instance
+	// will be specified by the endpoint's deployed model:
+	// https://storage.googleapis.com/google-cloud-aiplatform/schema/predict/instance/vision_embedding_model_1.0.0.yaml
+	instance, err := structpb.NewValue(map[string]any{
+		"image": map[string]any{
+			// Image input can be provided either as a Google Cloud Storage URI or as base64-encoded
+			// bytes using the "bytesBase64Encoded" field
+			"gcsUri": "gs://cloud-samples-data/vertex-ai/llm/prompts/landmark1.png",
+		},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to construct request payload: %v", err)
+	}
+
+	req := &aiplatformpb.PredictRequest{
+		Endpoint:  endpoint,
+		Instances: []*structpb.Value{instance}, // The model supports only 1 instance per request
+	}
+
+	resp, err := client.Predict(ctx, req)
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate embeddings: %v", err)
+	}
+
+	instanceEmbeddingsJson, err := protojson.Marshal(resp.GetPredictions()[0])
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert protobuf value to JSON: %v", err)
+	}
+	// Check the response schema of the model:
+	// https://storage.googleapis.com/google-cloud-aiplatform/schema/predict/prediction/vision_embedding_model_1.0.0.yaml
```

Comment:
Same as above: it looks like the YAML schema definition for the deployed model is the only documentation available.